pkg/remotemetrics/resources.go

```diff
@@ -0,0 +1,140 @@
+// Copyright 2019-2022 The Liqo Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package remotemetrics
+
+import (
+	"context"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/selection"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/klog/v2"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/liqotech/liqo/pkg/consts"
+	"github.com/liqotech/liqo/pkg/virtualKubelet/forge"
+)
+
+type resourceGetter struct {
+	cl client.Client
+}
+
+// NewResourceGetter creates a new resource getter.
+func NewResourceGetter(cl client.Client) ResourceGetter {
+	return &resourceGetter{
+		cl: cl,
+	}
+}
+
+// GetNamespaces returns the names of all namespaces in the cluster owned by the remote clusterID.
+func (m *resourceGetter) GetNamespaces(ctx context.Context, clusterID string) []MappedNamespace {
+	namespaces := &corev1.NamespaceList{}
+
+	err := m.cl.List(ctx, namespaces)
+	utilruntime.Must(err)
+
+	res := []MappedNamespace{}
+	for i := range namespaces.Items {
+		namespace := &namespaces.Items[i]
+
+		if getClusterIDFromNamespace(namespace) != clusterID {
+			continue
+		}
+		if namespace.Annotations == nil || namespace.Annotations[consts.RemoteNamespaceOriginalNameAnnotationKey] == "" {
+			continue
+		}
+
+		res = append(res, MappedNamespace{
+			Namespace:    namespace.GetName(),
+			OriginalName: namespace.Annotations[consts.RemoteNamespaceOriginalNameAnnotationKey],
+		})
+	}
+
+	klog.V(2).Infof("Scraping namespaces %+v for cluster id %s", res, clusterID)
+	return res
+}
+
+// GetPods returns the names of all pods in the cluster owned by the remote clusterID and scheduled in the given node.
+func (m *resourceGetter) GetPods(ctx context.Context, clusterID, node string) []string {
+	pods := &corev1.PodList{}
+
+	err := m.cl.List(ctx, pods)
+	utilruntime.Must(err)
```

Comment:
Would personally use the appropriate filters to always retrieve the desired pods.